src/etc/inc/certs.inc

```diff
@@ -269,6 +269,48 @@ function cert_create(&$cert, $caref, $keylen, $lifetime, $dn, $digest_alg = 'sha
     return true;
 }
 
+function sign_cert_for_csr(&$cert, $caref, $csr, $lifetime, $digest_alg, $x509_extensions = 'usr_cert')
+{
+    $ca = &lookup_ca($caref);
+    if (!$ca) {
+        return false;
+    }
+
+    $ca_str_crt = base64_decode($ca['crt']);
+    $ca_str_key = base64_decode($ca['prv']);
+    $ca_res_crt = openssl_x509_read($ca_str_crt);
+    $ca_res_key = openssl_pkey_get_private(array(0 => $ca_str_key, 1 => ""));
+    if (!$ca_res_key) {
+        return false;
+    }
+    $ca_serial = ++$ca['serial'];
+
+    $args = array(
```

Comment:
A certificate directly created from within OPNsense uses a temporary OpenSSL config based upon the template `/usr/local/etc/ssl/opnsense.cnf` to set a fixed and sane `keyUsage` and to handle `subjectAltName` tags, see `cert_create()` and the call to `create_temp_openssl_config()` therein.

Omitting the 'config' parameter for the `$args` array, as done here, makes OpenSSL use the config `/usr/local/openssl/openssl.cnf` instead. This and the missing handling of `subjectAltName` tags makes the generated certificate have the following issues:

1. Any X509v3 subjectAltName attribute values supplied in the CSR are dropped while they should indeed be copied 1:1

2. Any X509v3 Key Usage attributes supplied in the CSR are dropped while they should indeed be copied 1:1 *or* replaced by the fixed ones as specified in `opnsense.cnf`

Example X509v3 extensions from a CSR I just signed using the code in here:

- CSR
```
        Requested Extensions:
            X509v3 Basic Constraints: 
                CA:FALSE
            X509v3 Key Usage: 
                Digital Signature, Non Repudiation, Key Encipherment
            X509v3 Subject Alternative Name: 
                DNS:my.fqdn.tld
```
- Signed certificate:
```
        X509v3 extensions:
            X509v3 Basic Constraints: 
                CA:FALSE
            Netscape Comment: 
                OpenSSL Generated Certificate
            X509v3 Subject Key Identifier: 
                38:18:41:5C:9C:07:68:6F:DF:99:63:A2:3E:8E:42:3D:94:AB:3D:41
            X509v3 Authority Key Identifier: 
                keyid:38:01:5E:58:A1:1D:C2:4D:E6:62:03:8D:E1:EF:42:4F:57:BE:C2:94
```